backend/src/main/java/com/example/demo/todo/TodoService.java

```diff
@@ -17,21 +17,25 @@ public Todo createTodo(Todo todo) {
         return todoRepository.save(todo);
     }
 
-    public Collection<Todo> findAll() {
-        return todoRepository.findAll();
+    public List<Todo> getTodosbyUserEmail(String email) {
+        return todoRepository.findAllByUserEmail(email);
     }
 
     public Optional<Todo> findById(String id) {
             return todoRepository.findById(id);
         }
-    public Optional<Todo> updateTodo (String id, Todo updatedTodo) {
-        return todoRepository.findById(id)
+    public Optional<Todo> updateTodo (String id, String email, Todo updatedTodo) {
+        return todoRepository.findByIdAndUserEmail(id, email)
                 .map(todo -> todo.update(updatedTodo))
                 .map(todoRepository::save);
     }
 
-    public void deleteById(String id){
-        todoRepository.deleteById(id);
+    public Optional<Todo> deleteById (String id, String email){
+        Optional<Todo> todoToDelete = todoRepository.findByIdAndUserEmail(id, email);
+        if(todoToDelete.isPresent()) {
+            todoRepository.deleteById(id);
+        }
+        return todoToDelete;
```

Comment:
There is no feedback if the optional is empty.

Reply:
in Controller , but if you have any suggestion how to do it also in the Service, go ahead :)